### PR TITLE
Add queue name to HttpSysOptions

### PR DESCRIPTION
**Pull request by shirhatti**
- [x] Check requestQueueName for MAX_PATH
- [x] Figure out how to test queue name. There's no public API for retrieving Http.Sys queue name.

### CODE REVIEW

**Comment by shirhatti** [comment]
For https://github.com/aspnet/AspNetCore/issues/13461

**Comment by Tratcher** [comment]
> * [ ]  Figure out how to test queue name. There's no public API for retrieving Http.Sys queue name.

`netsh http show servicestate` and search the text for your guid queue name?

File: src/Servers/HttpSys/src/HttpSysOptions.cs [hunk 1]
```diff
@@ -23,11 +24,32 @@ public class HttpSysOptions
...
+        {
+            get => _requestQueueName;
+            set
+            {
+                if (value.Length > MaximumRequestQueueNameLength)
+                {
+                    throw new ArgumentException($"The request queue name should be fewer than {MaximumRequestQueueNameLength} characters in length",
+                                                nameof(value));
+                }
+                else
```
Comment (Tratcher) [inline src/Servers/HttpSys/src/HttpSysOptions.cs]: You can collapse the else since the if threw.

File: src/Servers/HttpSys/src/HttpSysOptions.cs [hunk 2]
```diff
@@ -23,11 +24,32 @@ public class HttpSysOptions
...
 
+        /// <summary>
+        /// The name of the Http.Sys request queue
+        /// </summary>
+        public string RequestQueueName
+        {
+            get => _requestQueueName;
+            set
+            {
+                if (value.Length > MaximumRequestQueueNameLength)
```
Comment (Tratcher) [inline src/Servers/HttpSys/src/HttpSysOptions.cs]: ArgumentOutOfRangeException

**Comment by Tratcher** [comment]
Or try creating a second instance with the same name and check that it throws a specific error code?

File: src/Servers/HttpSys/src/HttpSysOptions.cs [hunk 3]
```diff
@@ -23,11 +24,30 @@ public class HttpSysOptions
...
```
Comment (jkotalik) [inline src/Servers/HttpSys/src/HttpSysOptions.cs]: Confused why were are protecting against this. What happens if someone sets a request queue larger than 260 characters?

File: src/Servers/HttpSys/test/FunctionalTests/ServerTests.cs [hunk 4]
```diff
@@ -33,6 +34,30 @@ public async Task Server_200OK_Success()
...
+            {
+                options.RequestQueueName = queueName;
+            }))
+            {
+                var psi = new ProcessStartInfo("netsh", "http show servicestate view=requestq")
+                {
+                    RedirectStandardOutput = true
+                };
+                var process = Process.Start(psi);
+                process.Start();
```
Comment (jkotalik) [inline src/Servers/HttpSys/test/FunctionalTests/ServerTests.cs]: nit: Dispose the process.

Comment (Tratcher) [reply src/Servers/HttpSys/test/FunctionalTests/ServerTests.cs]: `using var process...`

File: src/Servers/HttpSys/test/FunctionalTests/Utilities.cs [hunk 5]
```diff
@@ -134,9 +141,8 @@ internal static IServer CreateDynamicHttpServer(string basePath, out string root
                     root = prefix.Scheme + "://" + prefix.Host + ":" + prefix.Port;
                     baseAddress = prefix.ToString();
 
-                    var server = CreatePump();
+                    var server = CreatePump(configureOptions);
```
Comment (jkotalik) [inline src/Servers/HttpSys/test/FunctionalTests/Utilities.cs]: nit: Can you remove `CreatePump()` with no arguments entirely and just use the new one? That way we don't get confused when to call configureOptions.

File: src/Servers/HttpSys/src/HttpSysOptions.cs (hunk 3 above)
Comment (shirhatti) [reply src/Servers/HttpSys/src/HttpSysOptions.cs]: > What happens if someone sets a request queue larger than 260 characters?

We know it'll fail (Windows MAX_PATH). This is just optimistically protecting against it.

Comment (Tratcher) [reply src/Servers/HttpSys/src/HttpSysOptions.cs]: What's the native error I'd get if I tried?

Comment (shirhatti) [reply src/Servers/HttpSys/src/HttpSysOptions.cs]: `ErrorCodes.ERROR_INVALID_PARAMETER`

File: src/Servers/HttpSys/src/HttpSysOptions.cs [hunk 6]
```diff
@@ -23,11 +24,30 @@ public class HttpSysOptions
         private RequestQueue _requestQueue;
         private UrlGroup _urlGroup;
         private long? _maxRequestBodySize = DefaultMaxRequestBodySize;
+        private string _requestQueueName;
```
Comment (jkotalik) [inline src/Servers/HttpSys/src/HttpSysOptions.cs]: Where is this value actually used? I feel like this change originally passed the request queue name somewhere but now it doesn't.

File: src/Servers/HttpSys/src/HttpSysOptions.cs (hunk 3 above)
Comment (jkotalik) [reply src/Servers/HttpSys/src/HttpSysOptions.cs]: Can someone change the max path for that parameter via a regkey? If so, I wouldn't throw here and just let HttpSys throw.

File: src/Servers/HttpSys/src/HttpSysOptions.cs (hunk 6 above)
Comment (Tratcher) [reply src/Servers/HttpSys/src/HttpSysOptions.cs]: Yeah, I think you lost something in the rebase.

File: src/Servers/HttpSys/test/FunctionalTests/Utilities.cs (hunk 5 above)
Comment (shirhatti) [reply src/Servers/HttpSys/test/FunctionalTests/Utilities.cs]: There are 10 other existing tests that use the existing CreatePump(). If I understand what you're asking for, you want me to replace all those use cases to pass around an empty delegate. Seems weird 🤷‍♂

Comment (jkotalik) [reply src/Servers/HttpSys/test/FunctionalTests/Utilities.cs]: Didn't realize it was called from many other places. Keep it then.

File: src/Servers/HttpSys/src/HttpSysOptions.cs (hunk 3 above)
Comment (shirhatti) [reply src/Servers/HttpSys/src/HttpSysOptions.cs]: Fair enough. I'll remove this check

**Review by Tratcher — APPROVED** [review]

**Pull request merged**